code/game/turfs/simulated/floor/plating/asteroid.dm

```diff
@@ -33,43 +35,20 @@
 
 /turf/open/floor/plating/asteroid/ex_act(severity, target)
 	contents_explosion(severity, target)
-	switch(severity)
-		if(3)
-			return
-		if(2)
-			if(prob(20))
-				src.gets_dug()
-		if(1)
-			src.gets_dug()
+	for(var/datum/component/archaeology/archy in datum_components)
```

Comment:
1. This loop is wrong, it should be
```
for(var/I in GetComponents(/datum/component/archaeology))
    var/datum/component/archaeology/archy = I
```
2. A better plan is to add a signal at the base of `ex_act()` and have the component itself listen for that signal and run this code